### PR TITLE
feat(sdk): openid4ci JS API.

### PR DESCRIPTION
**Pull request by vkubiv**

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: cmd/wallet-sdk-js/jsinterop/agent.go [hunk 1]
```diff
@@ -101,21 +104,36 @@ func CreateDID(_ js.Value, args []js.Value) (any, error) {
...
+			errors.InitializationFailedError,
+			fmt.Errorf("agent instance is not initialized"))
+	}
+
+	initiateIssuanceURI, err := jssupport.EnsureString(jssupport.GetNamedArgument(args, "initiateIssuanceURI"))
+	if err != nil {
+		return nil, err
+	}
+
+	interaction, err := agentInstance.CreateOpenID4CIInteraction(initiateIssuanceURI)
```
Comment (DRK3) [inline cmd/wallet-sdk-js/jsinterop/agent.go:124]: If my PR gets merged in first, can you rename this (and anything else referencing interactions) from `Interaction` to `IssuerInitiatedInteraction`? (If your PR gets merged in first, I will do the renaming)

**Review by birtony — APPROVED** [review]

File: cmd/wallet-sdk-js/jsinterop/types/openid4ci.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-js/jsinterop/jssupport"
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-js/walletsdk"
+	"github.com/trustbloc/wallet-sdk/pkg/models"
+)
+
+const (
+	openid4ciRequestCredentialWithPreAuth = "requestCredentialWithPreAuth"
+)
+
+func SerializeOpenID4CI(agentMethodsRunner *jssupport.AsyncRunner,
```
Comment (DRK3) [inline cmd/wallet-sdk-js/jsinterop/types/openid4ci.go:26]: Is this more like `SerializeOpenID4CIInteraction`?

Comment (DRK3) [reply cmd/wallet-sdk-js/jsinterop/types/openid4ci.go:26]: Or is it not exactly specific to an `Interaction`?

**Review by DRK3 — COMMENTED** [review]
Minor comments only

File: cmd/wallet-sdk-js/jsinterop/agent.go (hunk 1 above)
Comment (vkubiv) [reply cmd/wallet-sdk-js/jsinterop/agent.go:124]: yes, sure

Comment (DRK3) [reply cmd/wallet-sdk-js/jsinterop/agent.go:124]: Also, we'll need to add another one for the `WalletInitiatedInteraction`

File: cmd/wallet-sdk-js/jsinterop/types/openid4ci.go (hunk 2 above)
Comment (vkubiv) [reply cmd/wallet-sdk-js/jsinterop/types/openid4ci.go:26]: yes it more like SerializeOpenID4CIInteraction, will rename

File: cmd/wallet-sdk-js/jsinterop/agent.go (hunk 1 above)
Comment (vkubiv) [reply cmd/wallet-sdk-js/jsinterop/agent.go:124]: I think we will have a new issue for this.

Comment (vkubiv) [reply cmd/wallet-sdk-js/jsinterop/agent.go:124]: WalletInitiatedInteraction is out of the scope of this PR. Currently, we working on PreAuth only

**Pull request merged**